vignettes/validation_figure_tests.Rmd

```diff
@@ -0,0 +1,89 @@
+---
+title: "Validation: Figure Tests"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{validation_figure_tests}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(valtools)
+library(testthat)
+```
+
+Effective test cases cover a broad spectrum of use cases of the code intended to be validated. 
+More often than not, this is confirming that an input returns a specific or set of specific values.
+
+However, not all functions are used for their value-generation.
+In fact, there are a number of functions that are used for their side effects, such as plot generation. 
+This has long been the achilles heel of testing. 
+
+With {testthat} 3e (third edition), there came the advent to the R world the concept of snapshot testing. 
+That is, to create a reference file that is then compared against in future runs. 
+This is an incredibly powerful tool that creates the opportunity to add testing of not only known outputs in the form of numbers, but of files and figures as well.
+
+{valtools} is built around taking the brilliant work of {testthat} and using it to make validation easier. Here we will give a short demo on how to short-circuit these sorts tests to make our own figure validation paradigm.
+
+## Test Case
+
```

Comment:
make clear that this is {valtools} paradigm. perhaps mention that test cases and test code are both "validation elements"